client/app/containers/ConnectorManagement.jsx

```diff
@@ -187,6 +194,20 @@ export default class ConnectorManagement extends Component {
     return defaults
   }
 
+  checkProperties (connector) {
+    // remove invalid dataType declaration
+    // (valid = present in AUTHORIZED_DATATYPE)
+    let checkedTypes = []
+    for (let dType of connector.dataType) {
+      if (AUTHORIZED_DATATYPE.find(function (t) {
```

Comment:
I was looking at this PR, so I share my thought : `Array.find()` is a little bit expensive here for just testing an equality. We may use `Array.indexOf` instead. We may also use a function to avoid the comment `// (valid = present in AUTHORIZED_DATATYPE)`, like this one :
```javascript
const isValidType = (type) => AUTHORIZED_DATATYPE.indexOf(type) !== -1
```
We may also go a little further, because what we want to do in `checkProperties` is just __filter__ the valid dataTypes, and because we do not only _check_ but also update the connector, we may call the function _sanitize_ :
```javascript
sanitize (connector) {
  connector.dataType = connector.dataType.filter(isValidType)
  return connector
} 
```
Finally, to improve function purity and connector immutability, I think we may return another connector (I assume that it is ok with the way preact works:
```javascript
sanitize (connector){
  return Object.assign(
    {},
    connector,
    {
      dataType: connector.dataType.filter(isValidType)
    }
  }
```

What do you think ?